tasks/main.yml

```diff
@@ -1,5 +1,38 @@
 ---
 
+- name: check if spamd user exist
+  shell: "id -u {{ spamassassin_user }}"    
+  register: id_cmd
+  # rc=1 is OK, indicates that there is no spamd user
+  failed_when: id_cmd.rc >= 2
+
+- name: stop spamassassin
+  service:
+    name: spamassassin
+    state: stopped
+  when: id_cmd.rc == 0
+
+- name: Kill running processes for user spamd
+  shell: "killall -u {{ spamassassin_user }}"    
+  register: killall_cmd
+  # rc=1 is OK, indicates that there are no processes to be killed
+  failed_when: killall_cmd.rc >= 2
+  when: id_cmd.rc == 0
+
+- name: ensure spamd user is present
+  user:
+    name: "{{ spamassassin_user }}"
+    # Need to have a shell, otherwise updates in Debian fails
+    shell: /bin/sh
```

Comment:
Can you show a test case here? What update fails in Debian ? Aapt update ?

Reply:
It always works on a fresh install. I think it only happens when you upgrade or reinstall spamassassin. Here it happened apx 9 months after initial install (an autoupdate). 

To reproduce:

1: aptitude install spamassassin
2: usermod -s /bin/false debian-spamd
3: aptitude reinstall spamassassin

This article helped me debug it: 
https://askubuntu.com/questions/422264/how-to-debug-dpkg-configure-error-in-subprocess-post-installation/422697#422697